libhfcommon/util.c

```diff
@@ -235,7 +235,8 @@ static void util_rndInitThread(void) {
     while (rrnd < trnd) {
         // we are taking the harder road to cope with older linux versions without the getrandom's
         // wrapper
-        ssize_t len = (ssize_t)syscall(SYS_getrandom, (uint8_t*)rndState + rrnd, trnd - rrnd, 0);
+        ssize_t len = TEMP_FAILURE_RETRY((ssize_t)syscall(
+            SYS_getrandom, (uintptr_t)((uint8_t*)rndState + rrnd), (uintptr_t)(trnd - rrnd), 0));
```

Comment:
0L :)

Reply:
Also, this code is used directly by POSIX code, I think you need to leave the /dev/urandom path, and add getrandom as Linux specific (ifdef HF_ARCH_LINUX)